autosklearn/evaluation/abstract_evaluator.py

```diff
@@ -332,10 +336,14 @@ def file_output(
         # for one specific data set - optimization, validation or test!
         seed = self.seed
 
+        # Abort if self.Y_optimization is None
         # self.Y_optimization can be None if we use partial-cv, then,
         # obviously no output should be saved.
-        if self.Y_optimization is not None and \
-                self.Y_optimization.shape[0] != Y_optimization_pred.shape[0]:
+        if self.Y_optimization is None:
```

Comment:
I added this check, since the comment above said this was a possibility. 
Does this make sense? If yes, should we return an error like in the checks bellow?